content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -0,0 +1,331 @@
+
+This is an article about Sockets. It will give you a glimpse of how clients and servers communicate over the internet.
+
+Table of content:
+- [What is a socket](#what-is-a-socket)
+- [Programming on Server-side](#programming-on-server-side)
+- [](#)
+- [](#)
+- [](#)
+
+### Introduction
+Socket programming is a way of connecting two computers on a network to allow the exchange of information between the two machines that are running over `JRE`(Java Runtime Environment). The connection can be through **connection-oriented protocol** or **connectionless** but for our case, we will be using connection-oriented which is TCP/IP.
```

Comment:
> The connection can be through **connection-oriented protocol** or **connectionless** but for our case, we will be using connection-oriented which is TCP/IP. TCP is a connection-oriented protocol, which means that before data can be transferred, two computers must first establish a connection.

I would probably rephrase this to something like: 

"The connection can be through **connection-oriented protocol** or **connectionless**. In our case, we will be using TCP/IP, a connection-oriented protocol.	  
  
In connection-oriented protocols, two computers must establish a connection before transferring data."

> we will be using connection-oriented which is TCP/IP 

it sounds like you're saying that TCP/IP is synonymous to connection-orientated protocol but then you say: 

> TCP is a connection-oriented protocol

which says that it is a type of connection-oriented protocol